### ISSUE TITLE
Reading back empty spikeSourceArray with getSpikes() gives a numpy error.

### ISSUE THREAD

**Issue by SimonDavidson** [SpiNNakerManchester/sPyNNaker#155]
Created a spikeSourceArray with some spikes and the spikes were lost during simulation (as reported in an earlier bug report).

When I read back this array with getSpikes(), the fact that it was empty caused a fatal error in numpy:
2015-10-26 15:58:19 INFO: Getting spikes for excit_pop_ss_array
Getting spikes
|0                           50%                         100%|
 ============================================================
Traceback (most recent call last):
  File "multirun_single_neuron_stdp.py", line 19, in <module>
    weights = my_stdp.runModel(i+itOffset)
  File "/localhome/davidsos/tools/PyNNExamples/examples/libSingleNeuronStdp.py", line 162, in runModel
    stimSpikes =  populations[stimulus].getSpikes(compatible_output=True)
  File "/localhome/davidsos/tools/sPyNNaker/spynnaker/pyNN/models/pynn_population.py", line 215, in getSpikes
    compatible_output=compatible_output)
  File "/localhome/davidsos/tools/sPyNNaker/spynnaker/pyNN/models/spike_source/spike_source_array.py", line 497, in get_spikes
    result = numpy.vstack(results)
  File "/localhome/davidsos/tools/local/lib/python2.7/site-packages/numpy/core/shape_base.py", line 228, in vstack
    return _nx.concatenate([atleast_2d(_m) for _m in tup], 0)
ValueError: need at least one array to concatenate

**Comment by alan-stokes** [comment]
This is likely a issue with release version. This is ebcuase the spike soruce array no longer contains its own get_spikes with a numpy.vstack() function. Its been moved into a bespoke eieio_spiek_recorder. Its likely this may be resolved with a update to master.

**Comment by alan-stokes** [comment]
Having ran this script in the master version of the tool chain, Simon rubber stamped that the issue disappeared at that point. Therefore I'm recommending a pull update of Simon's software version with the master branches. I've updated his extra_models branch and commited the changes. 

Unfortantely it seems his software version on cspc026 has uncommited changes (whilst also stating that he's on the master branch), and to top it off, there's already a recurrent_update branch which hes not working on (which also has different updates to whats on his local machine).  

I've tried to stash, branch, and unstash on cscp276 on spynnaker to rectify the problem, but it seems that this has been refused for a number of unknown reasons. Until these are resolved, merging master back down to his software stack is beyond me.

Therefore I'm benching the upgrade till Rowley comes back and untangles the web of uncommitted changes to simons software version so that he can get up to speed with master.  

At which point this may all be rectifed and closed.
